src/main/scala/plex/PlexUtils.scala

```diff
@@ -60,9 +61,88 @@ trait PlexUtils {
       } yield result
     }.getOrElse(EitherT.left(IO.pure(new Throwable("Plex tokens are not configured"))))
 
+  protected def getOthersWatchlist(config: Configuration, client: HttpClient): EitherT[IO, Throwable, Set[Item]] =
+    for {
+      friends <- getFriends(config, client)
+      watchlistItems <- friends.map(getWatchlistIdsForUser(config, client)).toList.sequence.map(_.flatten)
+      items <- watchlistItems.map(i => toItems(config, client, i)).sequence.map(_.toSet)
+    } yield items
+
+  protected def getFriends(config: Configuration, client: HttpClient): EitherT[IO, Throwable, Set[User]] =
+    config.plexToken.map { token =>
+      val url = Uri
+        .unsafeFromString("https://community.plex.tv/api")
+
+      val query = GraphQLQuery(
+        """query GetAllFriends {
+          |        allFriendsV2 {
+          |          user {
+          |            id
+          |            username
+          |          }
+          |        }
+          |      }""".stripMargin)
+
+      EitherT(client.httpRequest(Method.POST, url, Some(token), Some(query.asJson)).map {
+        case Left(err) =>
+          logger.warn(s"Unable to fetch friends from Plex: $err")
+          Left(err)
+        case Right(json) =>
+          json.as[Users] match {
+            case Right(v) => Right(v.data.allFriendsV2.map(_.user).toSet)
+            case Left(v) => Left(new Throwable(v))
+          }
+      })
+    }.getOrElse(EitherT.left(IO.pure(new Throwable("Plex tokens are not configured"))))
+
+  protected def getWatchlistIdsForUser(config: Configuration, client: HttpClient)(user: User): EitherT[IO, Throwable, Set[TokenWatchlistItem]] =
+    config.plexToken.map { token =>
+      val url = Uri.unsafeFromString("https://community.plex.tv/api")
+
+      val query = GraphQLQuery(
+        """query GetWatchlistHub ($uuid: ID = "", $first: PaginationInt !, $after: String) {
+                user(id: $uuid) {
+                  watchlist(first: $first, after: $after) {
+                    nodes {
+                    ...itemFields
+                    }
+                    pageInfo {
+                      hasNextPage
+                      endCursor
+                    }
+                  }
+                }
+              }
+                fragment itemFields on MetadataItem {
+                id
+                title
+                type
+              }""".stripMargin,
+        Some(
+          s"""{
+             |  "first": 100,
+             |  "uuid": "${user.id}"
+             |}""".stripMargin.asJson))
+
+      EitherT(client.httpRequest(Method.POST, url, Some(token), Some(query.asJson)).map {
+        case Left(err) =>
+          logger.warn(s"Unable to fetch friends watchlist from Plex: $err")
+          Left(err)
+        case Right(json) =>
+          json.as[TokenWatchlistFriend] match {
+            // TODO: Fetch the other pages if hasNextPage = true
```

Comment:
Might release this first and then turn this todo into a ticket, will require additional testing etc.